src/utils.js

```diff
@@ -236,6 +236,84 @@ function camelcase (dashCaseText) {
     })
 }
 
+// We use this method to detect M1 macs and set appropriate API values to prevent sites from detecting fingerprinting protections
+function isAppleSilicon () {
+    const canvas = document.createElement('canvas')
+    const gl = canvas.getContext('webgl')
+
+    // Best guess if the device is an Apple Silicon
+    // https://stackoverflow.com/a/65412357
+    return gl.getSupportedExtensions().indexOf('WEBGL_compressed_texture_etc') !== -1
+}
+
+/**
+ * Take configSeting which should be an array of possible values.
+ * If a value contains a criteria that is a match for this environment then return that value.
+ * Otherwise return the first value that doesn't have a criteria.
+ *
+ * @param {array} configSetting - Config setting which should contain a list of possible values
+ * @returns - The value from the list that best matches the criteria in the config
+ */
+function processAttrByCriteria (configSetting) {
+    if (!configSetting.some(item => item.criteria !== undefined)) {
+        return configSetting
+    }
+
+    let bestOption = configSetting[0]
```

Comment:
This will fail if we have an erroneous _configSetting_ of an empty array. I think we should declare this as _undefined_ and the existing code will return it.
```suggestion
    let bestOption
```

In the calling code I think we should handle the _undefined_ case and return _defaultValue_